### ISSUE TITLE
Question about reward_loss in training/train_dpo.py

### ISSUE THREAD

**Issue by fengyang0317** [ZiyuGuo99/Image-Generation-CoT#22]
Thank for the work and sharing the code. I have a question regarding to the reward_loss. What is the reward_loss used for?  Which model is updated by the reward_loss? As far as I can I see, the backward of reward_loss cannot update any model. Do you use reinforcement learning in train_dpo.py?

[                    loss = unscaled_dpo_losses.mean() + config.training.reward_coef * reward_loss.mean()](https://github.com/ZiyuGuo99/Image-Generation-CoT/blob/main/training/train_dpo.py#L624)

**Comment by Wotoosh** [comment]
hi, do you have any ideas now? I have the same question as you, I don't quite understand the use of this loss used here, many thanks!
